driver/main.ml

```diff
@@ -156,11 +174,15 @@ let main () =
         (List.filter (fun x -> !x)
            [make_archive;make_package;compile_only;output_c_object])
         > 1
-    then
-      if !print_types then
-        fatal "Option -i is incompatible with -pack, -a, -output-obj"
-      else
+    then begin
+      let module P = Clflags.Compiler_pass in
+      match !stop_after with
+      | None ->
         fatal "Please specify at most one of -pack, -a, -c, -output-obj";
+      | Some (P.Parsing | P.Typing) ->
+          fatal "Options -i and -stop-after (parsing|typing)\
```

Comment:
I suggest to use something like `String.concat "|" P.names` rather than hard-code the list in various places. This way things are more likely to stay in sync when we add more constructors to `P.t`.

Reply:
We still have those present in the documentation, which is not procedurally generated. But your idea works well in combination with `Arg.Symbol`, will do, thanks.